pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4861,6 +4861,104 @@ int BPF_KPROBE(trace_ret_exec_binprm2)
+
+    // Get mapped memory information
+    struct file *file = get_mapped_file_from_vma(vma);
+    file_info = get_file_info(file);
+    // If there is no file that is mapped to the vma, try to get memory location name instead.
+    if (!file_info.pathname_p) {
+        get_vma_location(vma, sizeof(location), location);
+    }
+
+    struct task_struct *task = get_owner_task_from_mm(mm);
+    u32 pid = get_task_host_tgid(task);
+
+    save_to_submit_buf(&p->event->args_buf, &pid, sizeof(u32), 0);
+    save_to_submit_buf(&p->event->args_buf, &address, sizeof(void *), 1);
+    save_to_submit_buf(&p->event->args_buf, &gup_flags, sizeof(unsigned int), 2);
+    save_to_submit_buf(&p->event->args_buf, &vm_flags, sizeof(unsigned long), 3);
+    if (file_info.pathname_p != NULL) {
+        save_str_to_buf(&p->event->args_buf, file_info.pathname_p, 4);
+        save_to_submit_buf(&p->event->args_buf, &file_info.id.device, sizeof(dev_t), 5);
+        save_to_submit_buf(&p->event->args_buf, &file_info.id.inode, sizeof(unsigned long), 6);
+        save_to_submit_buf(&p->event->args_buf, &file_info.id.ctime, sizeof(u64), 7);
+    } else {
+        save_str_to_buf(&p->event->args_buf, location, 4);
+    }
+
+    return events_perf_submit(p, ACCESS_REMOTE_VM, 0);
+}
+
+SEC("kprobe/generic_access_phys")
+int BPF_KPROBE(trace_generic_access_phys)
+{
+    program_data_t p = {};
+    if (!init_program_data(&p, ctx))
+        return 0;
+
+    if (!should_trace(&p))
+        return 0;
+
+    struct vm_area_struct *vma = (struct vm_area_struct *) PT_REGS_PARM1(ctx);
+    unsigned long addr = (unsigned long) PT_REGS_PARM2(ctx);
+    int write = (int) PT_REGS_PARM5(ctx);
+    return submit_access_remote_vm(&p, vma, (void *) addr, write);
+}
+
+SEC("kprobe/get_user_pages_remote")
+TRACE_ENT_FUNC(get_user_pages_remote, ACCESS_REMOTE_VM);
+
+SEC("kretprobe/get_user_pages_remote")
+int BPF_KPROBE(trace_ret_get_user_pages_remote)
+{
+    args_t saved_args;
+    if (load_args(&saved_args, ACCESS_REMOTE_VM) != 0) {
+        // missed entry or not traced
+        return 0;
+    }
+    del_args(ACCESS_REMOTE_VM);
+
+    program_data_t p = {};
+    if (!init_program_data(&p, ctx))
+        return 0;
+
+    if (!should_trace(&p))
+        return 0;
+
+    int ret_val = PT_REGS_RC(ctx);
+    // if failed, then the operation is done using vm_ops->acess instead and will be traced by it
+    if (ret_val <= 0)
+        return 0;
+
+    unsigned long address = (unsigned long) saved_args.args[1];
+    unsigned int gup_flags = (unsigned int) saved_args.args[3];
+    struct vm_area_struct **vmas = (struct vm_area_struct **) saved_args.args[5];
+    struct vm_area_struct *vma = NULL;
+    unsigned long vm_flags = 0;
+    file_info_t file_info = {};
+    // Without a vma we can't get enough information for the event.
+    // Moreover, this will filter out the calls for it from the execve flow.
+    if (!vmas)
+        return 0;
+
+    bpf_core_read(&vma, sizeof(void *), vmas);
```

Comment:
You can mention that you're reading only the first VMA because `access_remote_vm->get_user_pages_remote` logic walks pages 1 by 1. It looks too detailed but it helps people with less experience in MM & VMA to understand why you're doing what you're doing.